CASMPET-5593 Update spire postgres without backup restore doc (#1635)
Cray-HPE/docs-csm: operations/spire/Restore_Spire_Postgres_without_a_Backup.md

   ```

2. Wait for the pods in the Spire namespace to terminate. Once that is done, remove
   the spire-data-server `PVCs`.

   ```bash
   ncn# kubectl get pvc -n spire | grep spire-data-spire-server | awk '{print $1}' | xargs kubectl delete -n spire pvc
   ```

> 3. Disable spire-agent on all of the Kubernetes NCNs (all worker nodes and

Reviewer: ```suggestion
3. Disable `spire-agent` on all of the Kubernetes NCNs (all worker nodes and
```

   master nodes) and delete the join data.

   ```bash
   ncn# for ncn in $(kubectl get nodes -o name | cut -d'/' -f2); do ssh "${ncn}" systemctl stop spire-agent; ssh "${ncn}" rm /root/spire/data/svid.key /root/spire/agent_svid.der /root/spire/bundle.der; done
   ```

## Re-install the Spire Helm Chart

1. Get the current cached customizations.

   ```bash
   ncn# kubectl get secrets -n loftsman site-init -o jsonpath='{.data.customizations\.yaml}' | base64 -d > customizations.yaml
   ```

1. Get the current cached `sysmgmt` manifest.

   ```bash
   ncn# kubectl get cm -n loftsman loftsman-sysmgmt -o jsonpath='{.data.manifest\.yaml}' > spire.yaml

> 

Reviewer: ```suggestion
```

   ncn# yq w -i spire.yaml metadata.name spire
   ncn# yq d -i spire.yaml spec.sources
   ncn# yq w -i spire.yaml spec.sources.charts[0].location 'https://packages.local/repository/charts'
   ncn# yq w -i spire.yaml spec.sources.charts[0].name csm-algol60
   ncn# yq w -i spire.yaml spec.sources.charts[0].type repo
   ```

   Example `spire.yaml` after the command is run:

   ```yaml
   apiVersion: manifests/v1beta1
     metadata:
       name: spire
     spec:
       charts:
         - name: spire
           namespace: spire
           source: csm-algol60
           values:
             server:
               fqdn: spire.local
             trustDomain: shasta
           version: 0.11.5
       charts:
         - location: https://packages.local/repository/charts
           name: csm-algol60
           type: repo
   ```

1. Generate the manifest that will be used to redeploy the chart with the
   modified resources.

   ```bash
   ncn# manifestgen -c customizations.yaml -i spire.yaml -o manifest.yaml
   ```

1. Validate that the `manifest.yaml` file only contains chart information for
   Spire, and that the sources chart location points to

>    `https://packages.local/repository/charts`

Reviewer: ```suggestion
   `https://packages.local/repository/charts`.
```

> 3. Get the current cached sysmgmt manifest.

Reviewer: ```suggestion
3. Get the current cached `sysmgmt` manifest.
```

   ```bash
   ncn# kubectl get cm -n loftsman loftsman-sysmgmt  -o jsonpath='{.data.manifest\.yaml}' > spire.yaml

   ```

> 4. Run the following command to remove non-spire charts from the spire.yaml file. This will also change the metadata.name so that it doesn't overwrite the sysmgmt.yaml file that is stored in the loftsman namespace.

Reviewer: ```suggestion
4. Run the following command to remove non-Spire charts from the `spire.yaml` file.

   This will also change the `metadata.name` so that it does not overwrite the `sysmgmt.yaml` file that is stored in the `loftsman` namespace.
```

   ```bash

>    for i in $(yq r spire.yaml 'spec.charts[*].name' | grep -Ev '^spire$'); do yq d -i spire.yaml  'spec.charts(name=='"$i"')'; done; yq w -i spire.yaml metadata.name spire

Reviewer: ```suggestion
   ncn# for i in $(yq r spire.yaml 'spec.charts[*].name' | grep -Ev '^spire$'); do
                yq d -i spire.yaml  'spec.charts(name=='"$i"')'
            done ; yq w -i spire.yaml metadata.name spire
```

   ```

>    Example spire.yaml after the command is run:

Reviewer: ```suggestion
   Example `spire.yaml` after the command is run:
```

>    ```text

Reviewer: ```suggestion
   ```yaml
```

   apiVersion: manifests/v1beta1
     metadata:
       name: spire
     spec:
       charts:
         - name: spire
           namespace: spire
           source: csm-algol60
           values:
             server:
               fqdn: spire.local
             trustDomain: shasta
           version: 0.11.5
   ```

5. Generate the manifest that will be used to redeploy the chart with the modified resources.

   ```bash
   ncn# manifestgen -c customizations.yaml -i spire.yaml -o manifest.yaml
   ```

> 6. Update the helm chart path in the manifest.yaml file.

Reviewer: ```suggestion
6. Update the helm chart path in the `manifest.yaml` file.
```

   ```

> 7. Validate that the manifest.yaml file only contains chart information for Spire,

Reviewer: ```suggestion
7. Validate that the `manifest.yaml` file only contains chart information for Spire,
```

>    and that the sources charts location points to the directory the helm chart was extracted from to prepend to /helm.

Reviewer: I don't know if this is what the sentence meant, but as it was written, it did not make much sense. 
```suggestion
   and that the source charts location points to the directory that the helm chart was extracted from.
```
